dallinger/config.py

```diff
@@ -346,20 +346,20 @@ def initialize_experiment_package(path):
     init_py = os.path.join(path, "__init__.py")
     if not os.path.exists(init_py):
         open(init_py, "a").close()
-
     # Retain already set experiment module
     if sys.modules.get("dallinger_experiment") is not None:
         return
     dirname = os.path.dirname(path)
     basename = os.path.basename(path)
     sys.path.insert(0, dirname)
     package = __import__(basename)
-    if path not in package.__path__:
+    if str(path) not in str(package.__path__):
```

Comment:
What are `path` and `package.__path__`? Could this check be performed at a higher "semantic level" than string comparison? In other words, are we starting with intelligent objects that already support this sort of check, and then throwing that away to do a comparison using primitive values?

Reply:
The bug I found was that `path` ended up being a `Posix.Path` object and `package.__path__` ended up being a list of strings; the `in` check then failed incorrectly. Talking this through now, I'm realising that `str(package.__path__)` is quite ugly, returning something like this: `"['/Users/peter/git/psynet-package/psynet']"`. I didn't mean to write that -- my tendency to use vectorised languages is clearly showing!

I think I would update this now to the following:

```py
if str(path) not in package.__path__:
```

There are a few alternatives, I guess:

```py
from pathlib import Path
if path not in [Path(p) for p in package.__path__]:
```

or for type clarity:

```py
from pathlib import Path
if Path(path) not in [Path(p) for p in package.__path__]:
```

or indeed 

```py
if str(path) not in [str(p) for p in package.__path__]:
```

What do you think?

Reply:
I'm for the All Strings or the All Paths version (A or C). @silviot introduced `pathlib` use fairly recently, and having more usage would encourage further standardization, which might justify the added complexity here, but I don't feel strongly.

Reply:
OK, sounds good to me, I like the idea of C too. I'll do that now.